app/lib/katello/resources/candlepin.rb

```diff
@@ -174,12 +172,11 @@ def upstream_consumer_id
           end
 
           def upstream_consumer
-            @organization ||= Organization.current
-            fail _("Current organization not set.") unless @organization
-            @upstream_consumer = @organization.owner_details['upstreamConsumer']
-            fail _("Current organization has no manifest imported.") unless @upstream_consumer
+            fail _("Current organization not set.") unless Organization.current
+            upstream_consumer = Organization.current.owner_details['upstreamConsumer']
```

Comment:
The only downside I see (which also existed in the old way) is that the methods on this class using the upstream_consumer make numerous requests to local candlepin. Over time that could be a significant impact to the system - maybe it's a good idea to cache the owner_details on the org once we fetch it the first time?

Reply:
and you could use a lazy_accessor to do this easily